### PR TITLE
fence_vmware_soap: Use --login-timeout option

### PR DESCRIPTION
**Pull request by thanasis2028**
Fixes issue #446.

### CODE REVIEW

**Comment by oalbrigt** [comment]
ok to test

**Comment by oalbrigt** [comment]
retest this please

File: agents/vmware_soap/fence_vmware_soap.py [hunk 1]
```diff
@@ -57,7 +57,8 @@ def soap_login(options):
 
 	try:
 		headers = {"Content-Type" : "text/xml;charset=UTF-8", "SOAPAction" : "vim25"}
-		conn = Client(url + "/vimService.wsdl", location=url, transport=RequestsTransport(verify=verify), headers=headers)
+		login_timeout = int(options["--login-timeout"])
```
Comment (oalbrigt) [inline agents/vmware_soap/fence_vmware_soap.py:60]: This doesnt work with --disable-timeout=true (enabled by default when run from Pacemaker 2.0+). Can you add ` or 60` after the `int()` here, so it at least gets a good chance to succeed for that use-case?

Comment (thanasis2028) [reply agents/vmware_soap/fence_vmware_soap.py]: Ok, a check for `--disable-timeout` has been added.

File: agents/vmware_soap/fence_vmware_soap.py [hunk 2]
```diff
@@ -57,7 +57,8 @@ def soap_login(options):
 
 	try:
 		headers = {"Content-Type" : "text/xml;charset=UTF-8", "SOAPAction" : "vim25"}
-		conn = Client(url + "/vimService.wsdl", location=url, transport=RequestsTransport(verify=verify), headers=headers)
+		login_timeout = 60 if "--disable-timeout" in options and options["--disable-timeout"] != "false" else int(options["--login-timeout"])
```
Comment (oalbrigt) [inline agents/vmware_soap/fence_vmware_soap.py:60]: It's easier to just do `or 60` like here:
https://github.com/ClusterLabs/fence-agents/blob/master/agents/gce/fence_gce.py#L296

As disable-timeout can be other true/false values as well.

Comment (thanasis2028) [reply agents/vmware_soap/fence_vmware_soap.py]: Ok, I now understand. My bad.

**Comment by thanasis2028** [comment]
Note that this timeout only seems to apply to the initial connection attempt, at least with my setup. Any rpc requests can still take too long to get a reply, and cause the pacemaker to kill it (with SIGKILL). In that case, any temp files created will remain forever, as described in #446.

File: agents/vmware_soap/fence_vmware_soap.py (hunk 2 above)
Comment (oalbrigt) [reply agents/vmware_soap/fence_vmware_soap.py]: No worries :).

**Comment by oalbrigt** [comment]
Thanks.

**Pull request merged**